src/main/deltachat/messagelist.ts

```diff
@@ -198,23 +198,23 @@ export default class DCMessageList extends SplitOut {
         if (chatId === -1) {
           chatId = message.chatId
         }
-        if (
-          getDirection(message) === 'incoming' &&
-          message.state !== C.DC_STATE_IN_SEEN
-        ) {
-          markMessagesRead.push(messageId)
+        // mark messages read triggers various mechanics in the core
+        // and should always be triggered if the user displays those
+        // messages on the screen.
+        if (getDirection(message) === 'incoming') {
+          markMessagesSeen.push(messageId)
```

Comment:
I don't like this logic in the backend, also could this be responsible for the new startup lag?

Reply:
It's probably fast as there is no additional database query, but even checking direction is unnecessary. It's completely ok to just call markSeenMessages for all messages displayed, including outgoing.